src/pseudopeople/configuration/generator.py

```diff
@@ -103,7 +104,8 @@ def get_configuration(
         is_no_noise = False
     noising_configuration = _generate_configuration(is_no_noise)
     if overrides is not None:
-        add_overrides(noising_configuration, overrides, dataset_schema, user_filters)
+        overrides_dict: Dict = overrides
+        add_overrides(noising_configuration, overrides_dict, dataset_schema, user_filters)
```

Comment:
If `overrides` is guaranteed to be a `dict` when not none, why not this:
```python
if isinstance(overrides, Dict):
    add_overrides(...)
```

If this is not the case - and I think it isn't - then this typing is actually misleading, since it may be the case that we get a list here.

Perhaps the right thing to do here is remove `validate_overrides` from insude `add_overrides` and do this:
```python
if overrides is not None:
    validate_overrides(overrides, noising_configuration)
    add_overrides(...)
```

Reply:
>  since it may be the case that we get a list here.
Since we're loading from yaml here, is there actually a limit to what basic python types you could technically get out of this? I dont know what constraints we have on the yaml format upstream.

How should I type `validate_overrides` in this situation?